src/lib/coda_lib/coda_subscriptions.ml

```diff
@@ -78,7 +79,35 @@ let create ~logger ~wallets ~time_controller ~external_transition_database
                 , `String (Staged_ledger.Pre_diff_info.Error.to_string e) ) ] ;
           Deferred.unit )
   |> don't_wait_for ;
-  {subscribed_payment_users; subscribed_block_users}
+  let reorganization_subscription = [] in
+  let reader, writer =
+    Strict_pipe.create ~name:"Reorganization subscription"
+      Strict_pipe.(Buffered (`Capacity 1, `Overflow Drop_head))
+  in
+  let t =
+    { subscribed_payment_users
+    ; subscribed_block_users
+    ; reorganization_subscription }
+  in
+  don't_wait_for
+  @@ Broadcast_pipe.Reader.iter transition_frontier
+       ~f:
+         (Option.value_map ~default:Deferred.unit
+            ~f:(fun transition_frontier ->
+              let best_tip_diff_pipe =
+                Transition_frontier.best_tip_diff_pipe transition_frontier
+              in
+              Broadcast_pipe.Reader.iter best_tip_diff_pipe
+                ~f:(fun {removed_user_commands; _} ->
+                  if List.is_empty removed_user_commands then
```

Comment:
Why is this condition here? All writes to the `best_tip_diff` indicate a reorg, right? I guess except the first one... https://github.com/CodaProtocol/coda/blob/master/src/lib/transition_frontier/diff.ml#L322-L338

Reply:
We're taking this definition for reorg https://en.bitcoin.it/wiki/Chain_Reorganization 

Best tip changes that just extend the path should not trigger a reorg

Reply:
Best tip changes that just extend the path don't get written to the `best_tip_diff_pipe`.

Reply:
I see yeah, I guess @johnwu93 did the is_empty check here to exclude the first case. That is correct, though, we don't want this to fire after we get a new frontier either on startup or post-bootstrap.

Reply:
For posterity: another problem with this approach is that it will fail to notice reorgs to branches that consist only of coinbases, which will be common during periods of low activity.